src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        checkAddIndex(index);
+        if (index == size) {
+            link(value);
+        } else {
```

Comment:
in your case it is always false
```suggestion
        if (index == size) {
            add(value);
            return;
        }
        checkAddIndex(index);
```